### ISSUE TITLE
Automatically provide default run configurations after opening a project for the first time

### ISSUE THREAD

**Issue by ortem** [intellij-rust/intellij-rust#9346]
## Problem description

It would be nice to automatically provide some default run configurations after opening a Rust project in the IDE for the very first time or after cloning a new project from VCS.  Such default configurations could include one `cargo run` configuration per [Cargo target](https://doc.rust-lang.org/cargo/reference/cargo-targets.html), as well as potentially `cargo test` configuration(s). Other possibilities could also be investigated and discussed.

**Comment by afetisov** [comment]
What would you suggest it should look for Cargo workspaces? These can have a huge number of targets.

Perhaps it would be better to provide an easy way to create a Run Configuration from a target in the Cargo panel.

**Comment by Undin** [comment]
> What would you suggest it should look for Cargo workspaces? These can have a huge number of targets.

Of course, it requires some investigation but if we are talking about some application with runnable project - how many `main` function (except examples) does an average project contain?

> Perhaps it would be better to provide an easy way to create a Run Configuration from a target in the Cargo panel.

You already can. Just make a double click on target item

**Comment by afetisov** [comment]
> Just make a double click on target item

That would run the target. It doesn't allow me to e.g. set some env variables or cmdline arguments before running it. Of course, one can always abort the process, but that's arguably poor UX, and can potentially have undesired side effects (maybe the binary without parameters does `rm -rf`). The target is also a temporary.

> how many main functions (except examples) does an average project contain?

I don't know whether that's average, but my monorepo project currently contains 20 crates in a single workspace (we're doing some dependency vendoring & patching, so it's convenient to have a single Cargo.lock). Several of those crates are binaries. Half of the crates contain example and/or test targets, and there can also be several integration tests or examples for a single crate.

Another example is [tokio](https://github.com/tokio-rs/tokio). I haven't counted, but it looks like there is a ton of crates and integration tests in that workspace.

I assume big projects will gravitate towards the same model, if they use a monorepo.

If you suggest to exclude all test and example targets from default run configurations, well, [solana](https://github.com/solana-labs/solana) has 29 `main.rs` files in its repo, none of them obviously excludable.

**Comment by afetisov** [comment]
Another good one: [tonic](https://github.com/hyperium/tonic). Its tests and examples are included as ordinary workspace crates which can not be autodetected as test/example executables by Cargo. There are 47 example binary crates and 11 test crates.

**Comment by afetisov** [comment]
Just a note, I'm not against the default run configurations in general. The current situation where a newly opened project doesn't have any configurations at all isn't good UX either. I would expect to see at least `cargo run` and `cargo test` for a new project.

My point is that simply dumping `cargo run --bin <bin>` for all `<bin>` binaries isn't really a scalable solution. Maybe we can use it in some simple cases, like a single-crate workspace or when the number of targets is small enough, but the general case needs more design work.
